app/models/user.rb

```diff
@@ -314,6 +314,36 @@ def compressed_export
     ActiveSupport::Gzip.compress Diaspora::Exporter.new(self).execute
   end
 
+  ######### Photos export ##################
+  mount_uploader :exported_photos_file, ExportedPhotos
+
+  def queue_export_photos
+    update exporting_photos: true
+    Workers::ExportPhotos.perform_async(id)
+  end
+
+  def perform_export_photos!
+    temp_zip = Tempfile.new([username, '_photos.zip'])
+    Zip::ZipOutputStream.open(temp_zip.path) do |zos|
+      photos.each do |photo|
+        begin
+          photo_data = photo.unprocessed_image.file.read
```

Comment:
This is what a thought:
- If I delete a post with a photo in it, photo file is deleted from my disc.
- I don't found code that delete (unlink) the file. So there's an object that knows what to do.
- The uploader field may know where to upload and to delete it.
- If so, it may also know where to read it.
- I tried it and worked, at least in my local file system.
- I found that admin can choose where to store files by setting [carrierwave config file](https://github.com/diaspora/diaspora/blob/develop/config/initializers/carrierwave.rb). I didn't try over S3. Amazon wants my credit card number for a free account and... no, darling.

What I really want to know is how to create a test to simulate a S3 storage. Do you have a sample code?